src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalProductOpt.java

```diff
@@ -0,0 +1,109 @@
+    private static final String BUNDLE_SUFFIX = ".json";
+
+    private BaseDTO dto = null;
+
+    public LocalProductOpt(BaseDTO dto) {
+        this.dto = dto;
+    }
+
+    public List<String> getSupportedLocales() {
+        List<String> supportedLocales = new ArrayList<String>();
+        try {
+
+            Path path = Paths.get(VIPCfg.getInstance().getOfflineResourcesBaseUrl());
+
+            URI uri = Thread.currentThread().getContextClassLoader().
+                    getResource(path.toString()).toURI();
+
+            if (uri.getScheme().equals("jar")) {
+                try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
+                    path = fileSystem.getPath(path.toString());
+                    getSupportedLocales(path, supportedLocales);
+                }
+            } else {
+                path = Paths.get(uri);
+                getSupportedLocales(path, supportedLocales);
+            }
+
+        } catch (Exception e) {
+            logger.debug(e.getMessage());
+        }
+        return supportedLocales;
+    }
+
+    public List<String> getComponents() {
+        List<String> components = new ArrayList<String>();
+        try {
+
+            Path path = Paths.get(VIPCfg.getInstance().getOfflineResourcesBaseUrl());
+
+            URI uri = Thread.currentThread().getContextClassLoader().
+                    getResource(path.toString()).toURI();
+
+            if (uri.getScheme().equals("jar")) {
+                try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
+                    path = fileSystem.getPath(path.toString());
+                    getComponents(path, components);
+                }
+            } else {
+                path = Paths.get(uri);
+                getComponents(path, components);
+            }
+
+        } catch (Exception e) {
+            logger.debug(e.getMessage());
+        }
+        return components;
+    }
+
+    private void getSupportedLocales(Path path, List<String> supportedLocales) throws IOException {
+        try (Stream<Path> listOfFiles = Files.walk(path).filter(p -> Files.isRegularFile(p))) {
+            listOfFiles.map(file -> {
+                String fileName = file.getFileName().toString();
+                if(fileName.startsWith(BUNDLE_PREFIX) && fileName.endsWith(BUNDLE_SUFFIX)) {
+                    return fileName.substring(BUNDLE_PREFIX.length(), fileName.indexOf('.'));
+                }
+                return "";
+            }).forEach(language -> {
+                if(language != null && !language.isEmpty()) {
+                    supportedLocales.add(language);
+                }
+            });
+        }
+    }
+
+    private void getComponents(Path path, List<String> components) throws IOException {
+        try (Stream<Path> listOfFiles = Files.walk(path).filter(p -> !Files.isRegularFile(p))) {
```

Comment:
Use Files.list here because Files.walk iterates recursively which we do not need here.
```suggestion
        try (Stream<Path> listOfFiles = Files.list(path).filter(p -> !Files.isRegularFile(p))) {
```